.github/workflows/main.yaml

```diff
@@ -194,6 +194,49 @@ jobs:
           cd ../..
           Rscript -e "devtools::test(reporter = c(\"summary\", \"fail\"))"
 
+  test_rust_client:
+    needs: [build_test_models]
+    runs-on: ${{matrix.os}}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+      fail-fast: false
+    steps:
+      - name: Check out github
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Restore Stan
+        uses: actions/cache@v3
+        with:
+          path: ./stan/
+          key: ${{ runner.os }}-stan-${{ hashFiles('stan/src/stan/version.hpp') }}-v${{ env.CACHE_VERSION }}
+
+      - name: Restore built models
+        uses: actions/cache@v3
+        id: test-models
+        with:
+          path: ./test_models/
+          key: ${{ hashFiles('**/*.stan', 'src/*', 'stan/src/stan/version.hpp') }}-${{ matrix.os }}-v${{ env.CACHE_VERSION }}
+
+      - name: Install LLVM and Clang  # Required for bindgen
+        uses: KyleMayes/install-llvm-action@v1
+        with:
+          version: "15.0"
+          directory: ${{ runner.temp }}/llvm
+
+      - name: Get rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Run rust tests
+        working-directory: ./rust
+        env:
+          LIBCLANG_PATH: ${{ runner.temp }}/llvm/lib
+          LLVM_CONFIG_PATH: ${{ runner.temp }}/llvm/bin/llvm-config
+        run: |
+          cargo test --verbose
```

Comment:
Should we do anything else besides a test here? e.g clippy or formatting

Reply:
I added clippy and fmt --check